lib/matplotlib/tests/test_collections.py

```diff
@@ -1101,3 +1101,13 @@ def test_set_offset_units():
     off0 = sc.get_offsets()
     sc.set_offsets(list(zip(y, d)))
     np.testing.assert_allclose(off0, sc.get_offsets())
+
+
+def test_set_offset_empty():
+    expected = mcollections.Collection(offsets=np.column_stack([[], []]))
+
+    modified = mcollections.Collection(
+        offsets=np.column_stack([[0, 1], [0, 1]])
+    )
+    modified.set_offsets([[], []])
```

Comment:
I think that this should actually be `column_stack` and I should get rid of the transpose in the set method. Most usage will be with `column_stack` i suspect.